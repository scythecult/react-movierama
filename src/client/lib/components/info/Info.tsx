import styles from './styles.module.css';

const getTotalPrice = (price: number, seats: []) => {
  const selectedCount = getSelectedCount(seats);

  return Number(price) * selectedCount || 0;
};

const getSelectedCount = (seats = []) => {
  if (seats.length) {
    return seats.reduce((initial, rows = []) => {
      const selected = rows.filter((seat) => seat);

      return (initial += selected.length);
    }, 0);
  }

  return 0;
};

export const Info = () => {
  const selectedCount = getSelectedCount();
  const totalPrice = getTotalPrice(10, []);

  return (
    <p className={styles.info}>
      You have selected <span>{selectedCount}</span> seats for a&nbsp;price&nbsp;of
      <span> ${totalPrice}</span>
    </p>
  );
};
