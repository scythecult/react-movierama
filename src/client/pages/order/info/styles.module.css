.info {
  padding: 5px 10px;
  text-align: center;
  border-radius: var(--app-border-radius-s);
  background-color: var(--app-safe-bg);
}

.infoText {
  color: var(--app-cyan-color);
}
