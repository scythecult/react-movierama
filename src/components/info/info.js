import React from "react";
import { useSelector } from "react-redux";

import classes from "./info.module.css";

const getTotalPrice = (price, seats) => {
  const { selectedMovie } = price;
  const selectedCount = getSelectedCount(seats);

  return Number(selectedMovie.price) * selectedCount || 0;
};

const getSelectedCount = (seats) => {
  if (seats?.seatsData) {
    return seats.seatsData.reduce((initial, rows) => {
      const selected = rows.filter((seat) => seat.isSelected);

      return (initial += selected.length);
    }, 0);
  }

  return 0;
};

const Info = () => {
  const { moviePicker, seats } = useSelector((state) => state);
  const selectedCount = getSelectedCount(seats);
  const totalPrice = getTotalPrice(moviePicker, seats);

  return (
    <p className={classes.info}>
      You have selected <span>{selectedCount}</span> seats for a&nbsp;price&nbsp;of
      <span> ${totalPrice}</span>
    </p>
  );
};

export { Info };
