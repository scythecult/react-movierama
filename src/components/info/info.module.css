.info {
  margin: 5px 0;
}

.info > span {
  color: #6feaf6;
}
