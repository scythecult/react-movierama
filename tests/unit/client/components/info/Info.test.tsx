import { render } from '@testing-library/react';
import { Info } from '@/client/lib/components/info/Info';

describe('Info', () => {
  test('should correspond default layout', () => {
    const result = render(<Info />);

    expect(result.container).toMatchSnapshot();
  });
});
