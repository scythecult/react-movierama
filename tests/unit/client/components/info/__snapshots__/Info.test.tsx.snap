// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Info > should correspond default layout 1`] = `
<div>
  <p
    class="_info_7377e4"
  >
    You have selected 
    <span>
      0
    </span>
     seats for a price of
    <span>
       $
      0
    </span>
  </p>
</div>
`;
